fix(learn): enhance exercise tracker user stories (#39965)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md

## Description
<section id='description'>
Build a full stack JavaScript app that is functionally similar to this: <a href='https://exercise-tracker.freecodecamp.rocks/' target='_blank'>https://exercise-tracker.freecodecamp.rocks/</a>. Working on this project will involve you writing your code using one of the following methods:

> - Clone <a href='https://github.com/freeCodeCamp/boilerplate-project-exercisetracker/'>this GitHub repo</a> and complete your project locally.

Reviewer: ```suggestion
- Clone <a href='https://github.com/freeCodeCamp/boilerplate-project-exercisetracker/' target='_blank'>this GitHub repo</a> and complete your project locally.
```

- Use <a href='https://repl.it/github/freeCodeCamp/boilerplate-project-exercisetracker' target='_blank'>our repl.it starter project</a> to complete your project.
- Use a site builder of your choice to complete the project. Be sure to incorporate all the files from our GitHub repo.

> When you are done, make sure a working demo of your project is hosted somewhere public. Then submit the URL to it in the `Solution Link` field. Optionally, also submit a link to your projects source code in the `GitHub Link` field.

Reviewer 2: ```suggestion
When you are done, make sure a working demo of your project is hosted somewhere public. Then submit the URL to it in the `Solution Link` field. Optionally, also submit a link to your project's source code in the `GitHub Link` field.
```

```yml
tests:
  - text: You should provide your own project, not the example URL.
    testString: |
      getUserInput => {
        const url = getUserInput('url');
        assert(!/.*\/exercise-tracker\.freecodecamp\.rocks/.test(getUserInput('url')));
      }

>   - text: You can `POST` to `/api/exercise/new-user` with form data `username` to create a new user. Returned will be `username` and `_id` keys.

Reviewer 2: ```suggestion
  - text: You can `POST` to `/api/exercise/new-user` with form data `username` to create a new user. Returned will be an object with `username` and `_id`.
```

>   - text: You can `POST` to `/api/exercise/new-user` with form data `username` to create a new user. Returned will be an object with `username` and `_id`.

Reviewer: ```suggestion
  - text: You can `POST` to `/api/exercise/new-user` with form data `username` to create a new user. The returned response will be an object with `username` and `_id` properties.
```

      }
    }
    "

>   - text: You can make a `GET` request to `api/exercise/users` to get an array of all users. Each item in the array is an object containing a users `username` and `_id`.

Reviewer: ```suggestion
  - text: You can make a `GET` request to `api/exercise/users` to get an array of all users. Each element in the array is an object containing a user's `username` and `_id`.
```

>   - text: You can make a `GET` request to `api/exercise/users` to get an array of all users. Each element in the array is an object containing a user's `username` and `_id`.

Reviewer 2: Just to mention it:
- Should we consider adding to the `testString` so that it matches the `text`/user-story more accurately?

      }
    }
    "

>   - text: 'You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date supplied it will use current date. Returned will be the user object with the exercise fields added.'

Reviewer: ```suggestion
  - text: 'You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date is supplied, it will use current date. The response returned will be the user object with the exercise fields added.'
```

>   - text: 'You can `POST` to `/api/exercise/add` with form data `userId(_id)`, `description`, `duration`, and optionally `date`. If no date supplied it will use current date. Returned will be the user object with the exercise fields added.'

Reviewer 2: ```suggestion
  - text: 'You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date supplied it will use current date. Returned will be the user object with the exercise fields added.'
```

>   - text: 'You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date is supplied, it will use current date. The response returned will be the user object with the exercise fields added.'

Reviewer 2: ```suggestion
  - text: 'You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date is supplied, the current date will be used. The response returned will be the user object with the exercise fields added.'
```

      }
    }
    "

>   - text: You can make a `GET` request to `/api/exercise/log` with a parameter of `userId=_id` to retrieve a full exercise log of any user. Returned will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date`. Also included is a `count` key that has the total exercise count.

Reviewer: ```suggestion
  - text: You can make a `GET` request to `/api/exercise/log` with a parameter of `userId=_id` to retrieve a full exercise log of any user. The returned response will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date` properties. Also included is a `count` key with the the total exercise count as its value.
```

>   - text: You can make a `GET` request to `/api/exercise/log` with a parameter of `userId=_id` to retrieve a full exercise log of any user. The returned response will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date` properties. Also included is a `count` key with the the total exercise count as its value.

Reviewer 2: Is there any reason the mention of the `count` key sounds like it is tested twice?

      }
    }
    "

>   - text: 'A request to `/api/exercise/log` includes a `count` key that is the number of exercises returned.'

Reviewer 2: See above ⏫

      }
    }
    "

>   - text: 'A request to `/api/exercise/log` includes a `count` key that is the number of exercises returned.'

Reviewer: The test is actually making a request to the following in the `testString`:

const logRes = await fetch(url + `/api/exercise/log?userId=${_id}`);

so I think it should be made clear that the log is for specified user referenced by a queryString parameter userid.

```suggestion
  - text: 'A request to a user's log (`/api/exercise/log) returns an object with `count` property representing the number of exercises for the user.'
```

>   - text: 'A request to a user's log (`/api/exercise/log`) returns an object with a `count` property representing the number of exercises returned.'

Reviewer 3: ```suggestion
  - text: "A request to a user's log (`/api/exercise/log`) returns an object with a `count` property representing the number of exercises returned."
```
This should resolve the lint error I believe.
